src/scvi/model/base/_archesmixin.py

```diff
@@ -326,7 +359,68 @@ def _get_loaded_data(reference_model, device=None):
     else:
         attr_dict = reference_model._get_user_attributes()
         attr_dict = {a[0]: a[1] for a in attr_dict if a[0][-1] == "_"}
-        var_names = reference_model.adata.var_names
+        if isinstance(reference_model.adata, MuData):
+            var_names = {
+                mod: reference_model.adata[mod].var_names
+                for mod in reference_model.adata.mod.keys()
+            }
+        else:
+            var_names = reference_model.adata.var_names
```

Comment:
this could be replaced by a call to `scvi.model.base._save_load._get_var_names`

Reply:
Done. Thanks for the pointer!